### ISSUE TITLE
pugixml can't be found

### ISSUE THREAD

**Issue by waarmond** [ornladios/ADIOS2#4081]
I'm trying since severals month to overcome the missing pugixml. The error occurs on both current git and 2.9.2
Since I'm on ArchLinux I would prefer to use system libs.

**Describe the bug**
```
CMake Error at source/adios2/toolkit/remote/CMakeLists.txt:12 (get_property):
  get_property could not find TARGET pugixml.  Perhaps it has not yet been
  created.
```

**To Reproduce**
```
  cmake -B build -S ADIOS2 \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_SKIP_RPATH=ON \
    -DADIOS2_USE_EXTERNAL_DEPENDENCIES=ON \
    -DADIOS2_HAVE_HDF5_VOL=OFF \
    -DADIOS2_BUILD_EXAMPLES=OFF
```
**Expected behavior**
configure, make, make install

**Desktop (please complete the following information):**
 - OS/Platform: ArchLinux
 - Build: gcc 13.2.1

**Comment by eisenhauer** [comment]
Interesting...  Obviously a CMake issue, but not having a handy ArchLinux platform sitting around, we're unlikely to be able to duplicate it (nor do we regularly test the use of external rather than supplied packages.). Perhaps if you included the output of CMake?  That would let us know if it actually found a system pugixml...

**Comment by waarmond** [comment]
Do you mean a config.log? I couldn't find one, but [here is the shell output](https://pastebin.com/HT58sGLy) on Pastebin.

**Comment by eisenhauer** [comment]
Probably needs to be looked at by our CMake people.  Are you looking at this because you want to use ADIOS or are you packaging?

**Comment by lahwaacz** [comment]
adios2 is [already packaged](https://gitlab.archlinux.org/archlinux/packaging/packages/adios2/-/blob/main/PKGBUILD?ref_type=heads) for Arch Linux and it links to system pugixml with no problems, as far as I can tell :shrug:

**Comment by waarmond** [comment]
@lahwaacz I've left the path of vanilla ArchLinux. For a git version of vtk I need a different version of adios2.

**Comment by eisenhauer** [comment]
@vicentebolea ?

**Comment by vicentebolea** [comment]
@waarmond thanks for sharing this with us.

Can you share your `cmakecache.txt`, if you do not have it, what is the pugixml version.

**Comment by chenrui333** [comment]
I am seeing the same while bumping adios2 to 2.10.0

```
CMake Error at source/adios2/toolkit/remote/CMakeLists.txt:12 (get_property):
  get_property could not find TARGET pugixml.  Perhaps it has not yet been
  created.


-- Looking for rdma/fi_ext_gni.h
-- Looking for rdma/fi_ext_gni.h - not found
CMake Error at source/utils/CMakeLists.txt:18 (get_property):
  get_property could not find TARGET pugixml.  Perhaps it has not yet been
  created.
```

- https://github.com/Homebrew/homebrew-core/pull/167957

**Comment by chenrui333** [comment]
2.9.2 worked well for homebrew builds before (including py3.12 build migration), the gcc 13.2.0 was used for both builds.

**Comment by cho-m** [comment]
2.10.0 issue was probably introduced in https://github.com/ornladios/ADIOS2/commit/dd432c15899040aee549b1c91849b0c18a482127

From quick glance, external `pugixml` target doesn't seem to be available in that scope (subdir `thirdparty/` vs `source/...`). Snippet of pugixml-config.cmake:
```cmake
if (NOT TARGET pugixml AND (NOT DEFINED PACKAGE_FIND_VERSION OR PACKAGE_FIND_VERSION VERSION_LESS "1.11"))
  add_library(pugixml INTERFACE IMPORTED)
  # Equivalent to target_link_libraries INTERFACE, but compatible with CMake 3.10
  set_target_properties(pugixml PROPERTIES INTERFACE_LINK_LIBRARIES pugixml::pugixml)
endif ()
```

I don't know what is recommended way, but some things that work are extending scope `find_package(pugixml CONFIG GLOBAL)` or using the other target `adios2::thirdparty::pugixml` which is `GLOBAL`.

**Comment by vicentebolea** [comment]
@cho-m thanks for the hints. It seems that to be a typo from our side, we must use the target adios2::thirdparty::pugixml within the adios2 source code. Can you try to replace pugixml to adios2::thirdparty::pugixml in `source/adios2/toolkit/remote/CMakeLists.txt:12` and check that it configures and builds?

**Comment by vicentebolea** [comment]
I believe that this was fixed at #4135, closing this please re-open if you still have the issue.
